platform/darwin/src/MGLRasterSource.h

```diff
@@ -1,93 +1,84 @@
-#import "MGLSource.h"
-#import "MGLTypes.h"
+#import "MGLTileSource.h"
 
 #import <CoreGraphics/CoreGraphics.h>
 
-@class MGLTileSet;
-
 NS_ASSUME_NONNULL_BEGIN
 
 /**
+ An `NSNumber` object containing a floating-point number that specifies the
+ width and height (measured in points) of each tiled image in the raster source.
+ 
+ This number also corresponds to the width and height (measured in points) at
+ which the map displays each tile when the map’s zoom level is an integer. The
+ source scales its images up or down when the map’s zoom level falls between two
+ integers.
+ 
+ This option applies only to tiles hosted by Mapbox and accessed through version
+ 4 of the
+ <a href="https://www.mapbox.com/api-documentation/#maps">Mapbox Maps API</a>,
+ such as through `mapbox://<mapid>` URLs. This option has no effect on tiles
+ accessed through the
+ <a href="https://www.mapbox.com/api-documentation/#static">Mapbox Static API</a>
+ or hosted on third-party tile servers.
+ 
+ The only supported values for this option are 256 and 512. When the value is
+ 256, the source requests 1× images on a standard-resolution display and 2×
+ images on a Retina display, and each image is scaled to fit in a square
+ measuring 256 points on each side. When the value is 512, the source requests
+ 2× images regardless of the screen resolution, and each image is displayed in a
+ square measuring 512 <em>pixels</em> on each side. If this option is set to a
+ value other than 256 or 512, the behavior is undefined.
+ 
+ The default value for this option is 256. This option is only applicable to
```

Comment:
This was a difficult option to document. I arrived at this verbose description of the `tileSize` option based on discussions with @jfirebaugh and @tmcw. For simplicity’s sake, I described only two valid values, 256 and 512, leaving all other values undefined.

MGLRasterTileSource applies a default value of 256 for mapbox: tile URLs and 512 for any other tile URL. This differs from the style specification and [GL JS](https://github.com/mapbox/mapbox-gl-js/blob/b5fce3485d8ed21a95bb77622b02fa1a874e3d2f/js/source/raster_tile_source.js#L20), which default to 512 regardless of origin. I believe MGLRasterTileSource’s behavior is justified, because the MGLRasterTileSource default and the underlying option in mbgl are applied only to mapbox: tile URLs, and mapbox: tile URLs are always normalized into Maps API v4 URLs.

Apparently we intend to revise the Maps API in the future to serve 512-point tiles (measuring 1 square kibipixel on a Retina display). A default value of 512 would make sense for requests to that API. However, I’d imagine that we’d map something other than `mapbox://<mapid>` to Maps API v5, in which case MGLRasterTileSource would automatically apply the standard 512 default, or we’d drop support for v4 tiles, which would be a backwards incompatible change anyways.

Meanwhile, at the moment, I believe a default other than 256 for mapbox: URLs would only lead to bugs in application code.